src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java

```diff
@@ -0,0 +1,134 @@
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class OnBehalfOfJwtAuthenticationTest {
+
+	public static final String POINTER_USERNAME = "/user_name";
+
+	static final TestSecurityConfig.User ADMIN_USER = new TestSecurityConfig.User("admin").roles(ALL_ACCESS);
+
+	private static final String signingKey = Base64.getEncoder().encodeToString("jwt signing key for an on behalf of token authentication backend for testing of OBO authentication".getBytes(StandardCharsets.UTF_8));
+	private static final String encryptionKey = Base64.getEncoder().encodeToString("encryptionKey".getBytes(StandardCharsets.UTF_8));
+	public static final String ADMIN_USER_NAME = "admin";
+	public static final String DEFAULT_PASSWORD = "secret";
+	public static final String OBO_TOKEN_REASON = "{\"reason\":\"Test generation\"}";
+	public static final String OBO_ENDPOINT_PREFIX = "_plugins/_security/api/user/onbehalfof";
+
+	@ClassRule
+	public static final LocalCluster cluster = new LocalCluster.Builder()
+			.clusterManager(ClusterManager.SINGLENODE).anonymousAuth(false)
+			.users(ADMIN_USER)
+			.nodeSettings(Map.of(
+					"plugins.security.allow_default_init_securityindex", true,
+					"plugins.security.restapi.roles_enabled", List.of("user_admin__all_access")
+			))
+			.authc(AUTHC_HTTPBASIC_INTERNAL)
+			.onBehalfOf(new OnBehalfOfConfig().signing_key(signingKey).encryption_key(encryptionKey))
+			.build();
+
+	@Test
+	public void shouldAuthenticateWithOBOTokenEndPoint() {
+		Header adminOboAuthHeader;
+
+		try (TestRestClient client = cluster.getRestClient(ADMIN_USER_NAME, DEFAULT_PASSWORD)) {
+
+			client.assertCorrectCredentials(ADMIN_USER_NAME);
+
+			TestRestClient.HttpResponse response = client.postJson(OBO_ENDPOINT_PREFIX, OBO_TOKEN_REASON);
+			response.assertStatusCode(200);
+
+			Map<String, Object> oboEndPointResponse = response.getBodyAs(Map.class);
+			assertThat(oboEndPointResponse, allOf(
+					aMapWithSize(3),
+					hasKey("user"),
+					hasKey("onBehalfOfToken"),
+					hasKey("duration")));
+
+			String encodedOboTokenStr = oboEndPointResponse.get("onBehalfOfToken").toString();
+
+			adminOboAuthHeader = new BasicHeader("Authorization", "Bearer " + encodedOboTokenStr);
+		}
+
+		try (TestRestClient client = cluster.getRestClient(adminOboAuthHeader)) {
+
+			TestRestClient.HttpResponse response = client.getAuthInfo();
+			response.assertStatusCode(200);
+
+			String username = response.getTextFromJsonBody(POINTER_USERNAME);
+			assertThat(username, equalTo(ADMIN_USER_NAME));
+		}
+	}
+
+	@Test
+	public void shouldNotAuthenticateWithATemperedOBOToken() {
```

Comment:
This is a good start, but we should aim to cover more scenarios in this suite. 

For this test why not try passing something known to be invalid like `invalid-token` instead of requesting a valid token from the endpoint and removing the last character?

Reply:
Cuz I would like to test if this token can be tampered based on this [comment](https://github.com/opensearch-project/security/pull/2857#discussion_r1236028482). But yeah, it would perform exactly the same of `invalid-token`.